servant-auth-server/src/Servant/Auth/Server/Internal/ConfigTypes.hs

```diff
@@ -26,26 +27,29 @@ data SameSite = AnySite | SameSiteStrict | SameSiteLax
   deriving (Eq, Show, Read, Generic, Ord)
 
 -- | @JWTSettings@ are used to generate cookies, and to verify JWTs.
-data JWTSettings = JWTSettings
-  {
-  -- | Key used to sign JWT.
-    signingKey      :: Jose.JWK
-  -- | Algorithm used to sign JWT.
-  , jwtAlg          :: Maybe Jose.Alg
-  -- | Keys used to validate JWT.
-  , validationKeys  :: Jose.JWKSet
-  -- | An @aud@ predicate. The @aud@ is a string or URI that identifies the
-  -- intended recipient of the JWT.
-  , audienceMatches :: Jose.StringOrURI -> IsMatch
-  } deriving (Generic)
+data JWTSettings
+  = JWTSettings
+      { -- | Key used to sign JWT.
+        signingKey :: Jose.JWK,
```

Comment:
In the Okta or Auth0 case, we wouldn't have a signing key. We can only validate using public keys. Should this signing key always be required? We could make a dummy one, of course...

Reply:
Yeah, it's a perpetual awkwardness of this library. The signing key is used for cookies (since those are also JWTs). There is sometimes an advantage to having it, since you can store data that's different than whatever you got during login, but it's really annoying to make it (and cookies) optional.

Reply:
Fair enough! Thanks again for working on this.